crates/flux-fhir-analysis/src/wf/sortck.rs

```
@@ -317,12 +337,25 @@ impl<'genv, 'tcx> InferCtxt<'genv, 'tcx> {
         }
     }
 
+    fn check_integral(&mut self, op: fhir::BinOp, sort: &rty::Sort, span: Span) -> Result {
+        if matches!(op, fhir::BinOp::Mod) {
+            let sort = self
+                .fully_resolve(sort)
+                .map_err(|_| self.emit_err(errors::CannotInferSort::new(span)))?;
+            if !matches!(sort, rty::Sort::Int | rty::Sort::BitVec(_)) {
+                span_bug!(span, "unexpected sort {sort:?} for operator {op:?}");
```

We should report an error here. You can trigger it with (I think)

```rust
#[sig(fn() requires 0real == 1real % 2real)]
fn foo() {}
```